blssig/cache_test.go

```diff
@@ -0,0 +1,51 @@
+package blssig
+
+import (
+	"runtime"
+	"testing"
+
+	bls12381 "github.com/drand/kyber-bls12381"
+	"github.com/drand/kyber/sign/bdn"
+	"github.com/stretchr/testify/require"
+
+	"github.com/filecoin-project/go-f3/gpbft"
+)
+
+const maxCacheMemory uint64 = 10 * 1023 * 1024 // 10MiB
+
+func TestCacheMemory(t *testing.T) {
+	suite := bls12381.NewBLS12381Suite()
+	scheme := bdn.NewSchemeOnG2(suite)
+
+	rand := suite.RandomStream()
+	keys := make([]gpbft.PubKey, maxPointCacheSize+1)
+	for i := range keys {
+		_, pub := scheme.NewKeyPair(rand)
+		pubKeyB, err := pub.MarshalBinary()
+		require.NoError(t, err)
+		keys[i] = pubKeyB
+	}
+
+	runtime.GC()
+	var beforeMemStats, afterMemStats runtime.MemStats
+	runtime.ReadMemStats(&beforeMemStats)
+	v := VerifierWithKeyOnG1()
+	for _, k := range keys[1:] {
+		_, err := v.pubkeyToPoint(k)
```

Comment:
Being picky: test `pubkeysToMask` too?

Reply:
That just calls `pubkeyToPoint` (all I'm testing here is that the cache doesn't grow too large). Everything else should already be tested by our signing test suite.